src/Bundle/Controller/Admin/ModuleRecommendedController.php

```diff
@@ -0,0 +1,89 @@
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/AFL-3.0  Academic Free License (AFL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\Module\Mbo\Bundle\Controller\Admin;
+
+use PrestaShop\Module\Mbo\Adapter\RecommendedModulePresenter;
+use PrestaShop\Module\Mbo\Core\Tab\TabCollectionProvider;
+use PrestaShopBundle\Controller\Admin\FrameworkBundleAdminController;
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpKernel\Exception\ServiceUnavailableHttpException;
+use Symfony\Component\Templating\EngineInterface;
+
+class ModuleRecommendedController extends FrameworkBundleAdminController
+{
+    /**
+     * @param Request $request
+     *
+     * @return JsonResponse
+     */
+    public function indexAction(Request $request)
+    {
+        $response = new JsonResponse();
+
+        try {
+            $tabCollectionProvider = $this->getTabCollectionProvider();
+            $tab = $tabCollectionProvider->getTab($request->get('tabClassName'));
+            $recommendedModulePresenter = new RecommendedModulePresenter();
+            $recommendedModulesInstalled = $tab->getRecommendedModulesInstalled();
+            $recommendedModulesNotInstalled = $tab->getRecommendedModulesNotInstalled();
+            $response->setData([
+                'content' => $this->getTemplateEngine()->render(
+                    '@Modules/ps_mbo/views/templates/admin/controllers/module_catalog/recommended-modules.html.twig',
+                    [
+                        'recommendedModulesInstalled' => $recommendedModulePresenter->presentCollection($recommendedModulesInstalled),
+                        'recommendedModulesNotInstalled' => $recommendedModulePresenter->presentCollection($recommendedModulesNotInstalled),
+                    ]
+                ),
+            ]);
+        } catch (ServiceUnavailableHttpException $exception) {
+            $response->setData([
+                'content' => $this->getTemplateEngine()->render('@Modules/ps_mbo/views/templates/admin/error.html.twig'),
+            ]);
+            $response->setStatusCode($exception->getStatusCode());
+            $response->headers->add($exception->getHeaders());
+        }
+
+        return $response;
+    }
+
+    /**
+     * @return TabCollectionProvider
+     */
+    private function getTabCollectionProvider()
+    {
+        return $this->get('mbo.tab.collection_provider');
+    }
+
+    /**
+     * @return EngineInterface
+     */
+    private function getTemplateEngine()
+    {
+        return $this->get('templating');
```

Comment:
you'd better use ``twig` service (more performant)

Reply:
Thanks :)